chore(docs): improve LATEST BY page

puzpuzpuz opened the pull request:
Includes the following changes:
* Renames LATEST BY to LATEST ON everywhere in the docs. Keeps a redirect for the old `/latest-by` page address.
* Adds a clarifying text for the LATEST ON intro text.
* Adds a new section for LATEST ON over sub-query.

bluestreak01 commented:
I feel the opening paragraph could better explain why we can't just use `LIMIT -1`.

>LATEST ON PARTITION BY is used as part of a [SELECT statement](https://deploy-preview-730--unruffled-blackwell-31bfb2.netlify.app/docs/reference/sql/select/) for returning the most recent records per unique column value, commonly on STRING and SYMBOL column types. For the sake of brevity we to refer to this clause as LATEST ON.

I'd like to suggest something along the lines of:

```
For scenarios where multiple time series are stored in the same table it is relatively difficult to identify last items of these time series with standard SQL syntax. We introduces `LATEST ON` extension for `SELECT` clause to remove boilerplate clutter and splice table with relative ease. 
```

WDYT?

bluestreak01 commented:
Directly after syntax railroad diagram we should attempt to document syntax components. In similar way method args are documents along side what method does.

bluestreak01 commented:
Here:

![image](https://user-images.githubusercontent.com/7276403/154471474-f7eb4f47-97b8-49b8-a86b-6aeb14e4ebf9.png)


Can we say "The old `LASTEST BY` syntax..." ?

bluestreak01 commented:
The document lacks description of how to time-slice, the last paragraph `LATEST ON first` does show mix of `where` clauses, but it is more powerful with time-slicing.

bluestreak01 commented:
We also should list all types, which can be used in `PARTITION BY` clause. I would also like to see description of how `LASTEST ON` will behave with non-symbol types and some advice around using timestamp to limit the performance impact

puzpuzpuz commented:
> The document lacks description of how to time-slice, the last paragraph `LATEST ON first` does show mix of `where` clauses, but it is more powerful with time-slicing.

Addressed all comments except for this one in 73439d10637bfca7f875fa8300be66c15a03b651

puzpuzpuz commented:
Addressed the remaining comment in 67ef3b743bd3da6ef3ca27151c6207a7cdc16ad3

bluestreak01 commented on docs/reference/sql/latest-on.md:
```diff
@@ -36,6 +51,16 @@ LATEST ON pickup_datetime PARTITION BY payment_type;
 | Card         | 2019-06-30T23:59:56.000000Z | 1             |
 
+The above query returns the latest value within each time series stored in the
+table. Those time series are determined based on the column(s) specified in the
```
```suggestion
table. Those time series are determined based on the values in the column(s) specified in the
```

puzpuzpuz replied:
Applied in 411eb87f12908c79fbcefdc59e3db3b97348f829

puzpuzpuz commented on docs/reference/sql/latest-on.md:
```diff
@@ -1,21 +1,36 @@
-records per unique column value, commonly on `STRING` and `SYMBOL` column types.
-For the sake of brevity we to refer to this clause as `LATEST BY`.
+For scenarios where multiple time series are stored in the same table it is
+relatively difficult to identify the lastest items of these time series with
```
No need for clarification, it's a typo. Thanks for spotting it. Fixed in 411eb87f12908c79fbcefdc59e3db3b97348f829
